File: nuido_flow_ai_mini_knowledge/flows/ai/mini_knowledge_memory_node.py
# THIS FILE IS A PART OF PUBLIC REPOSITORY https://github.com/yonitjio/exploring-odoo
#
# This software is released under the MIT License.
# https://opensource.org/licenses/MIT
#
# THIS SOFTWARE IS EXPERIMENTAL AND FOR EDUCATIONAL PURPOSE ONLY.
# DO NOT USE IT IN PRODUCTION.
from odoo.addons.nuido_flow.flows.core.base_node import BaseNode
from odoo.addons.nuido_flow_ai.flows.ai.memory_utils import create_memory

from .const import COLLECTION_NAME

class MiniKnowledgeMemoryNode(BaseNode):
    def _process(self, params):
        res = create_memory(COLLECTION_NAME)
        return {
            "memory": res
        }

File: nuido_flow_ai_mini_knowledge/static/src/models/ai/mini_knowledge_memory_node.js
/*!
// THIS FILE IS A PART OF PUBLIC REPOSITORY https://github.com/yonitjio/exploring-odoo
//
// This software is released under the MIT License.
// https://opensource.org/licenses/MIT
//
// THIS SOFTWARE IS EXPERIMENTAL AND FOR EDUCATIONAL PURPOSE ONLY.
// DO NOT USE IT IN PRODUCTION.
*/
import { NodeModel } from "@nuido/models/node";
import { AiMemoryPort } from "@nuido_flow_ai/components/ports/ai_memory_port";
export class MiniKnowledgeMemoryNodeModel extends NodeModel {
    setup() {
        const auxOutId = "aux-out-" + this.id + "-mini-knowledge-memory";
        this.addAuxOutPort(auxOutId, AiMemoryPort.name, Number.MAX_SAFE_INTEGER, {
            role: "ai-memory"
        });
    }
}
